var/spack/repos/builtin/packages/py-pip/package.py

```diff
@@ -30,6 +30,5 @@ class PyPip(PythonPackage):
     # However, pip requires setuptools during runtime as well.
     depends_on('py-setuptools', type=('build', 'run'))
 
-    def setup_environment(self, spack_env, run_env):
-        run_env.prepend_path('PATH', self.spec.prefix.bin)
-
+    def setup_run_environment(self, env):
```

Comment:
Curious : wasn't this already in the upstream ? i.e. how this was not done during initial merge with the upstream?

Reply:
Probably a merge artifact of ours.

Reply:
Checked again, not in upstream. IIRC, you or @jorblancoa added that?

Reply:
hmmm....Ok. I don't remember.